metrics-report-generator/src/main/scala/it/pagopa/interop/metricsreportgenerator/Main.scala

```diff
@@ -56,22 +59,35 @@ object Main extends App {
     blockingEC: ExecutionContextExecutor
   ): Future[Unit] = {
     val env: String = config.environment
-
-    val agreementsJobResult: Future[MailAttachment] = jobs.getAgreementRecord
-      .map(_.getBytes)
-      .flatMap(bs => s3.saveAgreementsReport(bs).map(_ => asAttachment(s"agreements-${env}.csv", bs)))
-
-    val activeDescriptorsJobResult: Future[MailAttachment] = jobs.getDescriptorsRecord
-      .map(_.getBytes)
-      .flatMap(bs => s3.saveActiveDescriptorsReport(bs).map(_ => asAttachment(s"active-descriptors-${env}.csv", bs)))
-
-    val tokensJobResult: Future[MailAttachment] = tokensJob.getTokensData
-      .map(_.getBytes)
-      .flatMap(bs => s3.saveTokensReport(bs).map(_ => asAttachment(s"tokens-${env}.csv", bs)))
-
-    Future
-      .sequence(List(agreementsJobResult, tokensJobResult, activeDescriptorsJobResult))
-      .flatMap(sendMail(config))
+    for {
+
+      _ <- jobs.getAgreementRecord
+        .map(_.getBytes)
+        .flatMap(bs => s3.saveAgreementsReport(bs))
+
+      _ <- jobs.getDescriptorsRecord
+        .map(_.getBytes)
+        .flatMap(bs => s3.saveActiveDescriptorsReport(bs))
+
+      _ <- tokensJob.getTokensDataCsv
+        .map(_.getBytes)
+        .flatMap(bs => s3.saveTokensReport(bs))
+
+      agreementsSheet  <- jobs.getAgreementSheet
+      descriptorsSheet <- jobs.getDescriptorsSheet
+      tokenSheet       <- tokensJob.getTokensDataSheet
```

Comment:
This way we access to S3 report only once (it may have a significant size)
```suggestion
     tokenReport <- tokensJob.getTokenReport()
      _ <- tokenReport.renderCsv
        .map(_.getBytes)
        .flatMap(bs => s3.saveTokensReport(bs))

      agreementsSheet  <- jobs.getAgreementSheet
      descriptorsSheet <- jobs.getDescriptorsSheet
      tokenSheet       <- tokenReport.renderSheet
```